pkg/runtimecompiler/runtimecompiler.go

```diff
@@ -0,0 +1,227 @@
+//go:generate go run ../ebpf/include_headers.go ../security/ebpf/c/prebuilt/probe.c ../ebpf/bytecode/build/runtime/runtime-security.c ../security/ebpf/c ../ebpf/c
+//go:generate go run ../ebpf/bytecode/runtime/integrity.go ../ebpf/bytecode/build/runtime/runtime-security.c ../ebpf/bytecode/runtime/runtime-security.go runtime
+
+//go:generate go run ../ebpf/include_headers.go ../collector/corechecks/ebpf/c/runtime/tcp-queue-length-kern.c ../ebpf/bytecode/build/runtime/tcp-queue-length.c ../ebpf/c
+//go:generate go run ../ebpf/bytecode/runtime/integrity.go ../ebpf/bytecode/build/runtime/tcp-queue-length.c ../ebpf/bytecode/runtime/tcp-queue-length.go runtime
+
+//go:generate go run ../ebpf/include_headers.go ../collector/corechecks/ebpf/c/runtime/oom-kill-kern.c ../ebpf/bytecode/build/runtime/oom-kill.c ../ebpf/c
+//go:generate go run ../ebpf/bytecode/runtime/integrity.go ..//ebpf/bytecode/build/runtime/oom-kill.c ../ebpf/bytecode/runtime/oom-kill.go runtime
+
+package runtimecompiler
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/ebpf/bytecode/runtime"
+	"github.com/DataDog/datadog-agent/pkg/runtimecompiler/config"
+	"github.com/DataDog/datadog-agent/pkg/security/ebpf"
+	seckernel "github.com/DataDog/datadog-agent/pkg/security/ebpf/kernel"
+	"github.com/DataDog/datadog-agent/pkg/security/probe"
+	"github.com/DataDog/datadog-agent/pkg/security/probe/constantfetch"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/version"
+	"github.com/DataDog/datadog-go/v5/statsd"
+)
+
+// RuntimeCompiler is responsible for runtime compiling eBPF programs across the system-probe.
+// Runtime compilation must be performed prior to any other modules being started in order
+// for those modules to be able to correctly load runtime compiled programs.
+var RuntimeCompiler = newRuntimeCompiler()
+
+const metricPrefix = "datadog.system_probe.runtime_compiler"
+
+type runtimeCompiler struct {
+	enabled      bool
+	config       *config.Config
+	statsdClient statsd.ClientInterface
+
+	headerFetchResult kernel.HeaderFetchResult
+	telemetryByAsset  map[string]runtime.RuntimeCompilationTelemetry
+}
+
+func newRuntimeCompiler() runtimeCompiler {
+	return runtimeCompiler{
+		enabled:           false,
+		config:            nil,
+		statsdClient:      nil,
+		headerFetchResult: kernel.NotAttempted,
+		telemetryByAsset:  make(map[string]runtime.RuntimeCompilationTelemetry),
+	}
+}
+
+// Init initializes the runtime compiler. This should be called prior to calling Run
+func (rc *runtimeCompiler) Init(cfg *config.Config, statsdClient statsd.ClientInterface) {
+	compilationRequired := cfg.EnableNetworkCompilation ||
+		cfg.EnableRuntimeSecurityCompilation ||
+		cfg.EnableConstantFetcherCompilation ||
+		cfg.EnableTcpQueueLengthCompilation ||
+		cfg.EnableOomKillCompilation
+
+	RuntimeCompiler.enabled = compilationRequired
```

Comment:
This should be `rc.enabled`? The affect is probably the same since the global looks like a singleton, but I think this would be more correct.

Reply:
yep, you're right! same goes for the other 2 things below it